app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -163,12 +202,44 @@ def parse_deployment_name(name)
         name.sub(/^.*deployment=/, '')
       end
 
-      def parse_middleware_server(eap)
+      def parse_server_group_name(name)
+        name.sub(/^Domain Server Group \[/, '').chomp(']')
+      end
+
+      def parse_domain_server_name(name)
+        name.sub(/^.*\/server=/, '')
+      end
+
+      def parse_middleware_domain(domain)
+        {
+          :feed       => domain.feed,
+          :ems_ref    => domain.path,
+          :nativeid   => domain.id,
+          :name       => domain.properties['Name'],
+          :type_path  => domain.type_path,
+          :properties => domain.properties
+        }
+      end
+
+      def parse_middleware_server_group(domain, group)
+        {
+          :feed              => group.feed,
+          :ems_ref           => group.path,
+          :nativeid          => group.id,
+          :middleware_domain => domain,
+          :name              => parse_server_group_name(group.name),
+          :type_path         => group.type_path,
+          :profile           => group.properties['Profile'],
+          :properties        => group.properties
+        }
+      end
+
+      def parse_middleware_server(eap, name = nil)
         {
           :feed       => eap.feed,
           :ems_ref    => eap.path,
           :nativeid   => eap.id,
-          :name       => parse_name(eap.id),
+          :name       => name || parse_name(eap.id),
```

Comment:
Because the method is called from 2 different contexts.

1. for the normal server running in the standalone mode (no `name` parameter - `parse_name(eap.id)` is being called)

2. for the server that is running in the domain mode, the name is passed as the parameter.

Oh, and the name has to be parsed differently for each context.